File: Linked_Lists/Medium/445/445.c
#include <stdio.h>
#include <stdlib.h>

// Definition for singly-linked list.
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    int s1[100];
    int s2[100];
    int s1Size = 0, s2Size = 0;
    while(l1 != NULL){
        s1[s1Size++] = l1->val;
        l1 = l1->next;
    }
    while(l2 != NULL){
        s2[s2Size++] = l2->val;
        l2 = l2->next;
    }
    int carry = 0;
    struct ListNode *head = NULL;
    while(s1Size || s2Size || carry){
        int d1 = s1Size ? s1[--s1Size] : 0;
        int d2 = s2Size ? s2[--s2Size] : 0;
        int val = d1 + d2 + carry;
        carry = val / 10;
        val = val % 10;
        struct ListNode *temp = (struct ListNode *)malloc(sizeof(struct ListNode));
        temp->next = head;
        temp->val = val;
        head = temp;
    }
    return head;
}

File: Linked_Lists/Medium/445/445.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        s1, s2 = [], []
        while l1:
            s1.append(l1.val)
            l1 = l1.next
        while l2:
            s2.append(l2.val)
            l2 = l2.next
        carry, head = 0, None
        while s1 or s2 or carry:
            d1 = s1.pop() if s1 else 0
            d2 = s2.pop() if s2 else 0
            val = d1 + d2 + carry
            carry, val = val // 10, val % 10
            head = ListNode(val, head)
        return head